### PR TITLE
Adds code to update delete site explanation when a site has MTM container, #PG-3574

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Adds code to update delete site explanation when a site has MTM container
Requires: https://github.com/matomo-org/matomo/pull/22484
Fixes: #PG-3574

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: templates/deleteWebsite.twig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<p>{{ 'TagManager_DeleteWebsiteExplanationLine1'|translate }}</p>
+<p>{{ 'TagManager_DeleteWebsiteExplanationLine2'|translate }}</p>
+<ol style="list-style-type:decimal !important;padding-left: 1rem;">
+    {% for container in containers %}
+        <li>
+            <a href="{{ link }}&idContainer={{ container.idcontainer }}" target="_blank" rel="noreferrer noopener">{{ container.name|e('html')|e('html') }}</a>
```
Comment (snake14) [inline templates/deleteWebsite.twig:6]: I'm guessing that this was an accidental double-paste?
```suggestion
            <a href="{{ link }}&idContainer={{ container.idcontainer }}" target="_blank" rel="noreferrer noopener">{{ container.name|e('html') }}</a>
```

File: tests/UI/expected-screenshots/TagManager_manageWebsitesDeleteAction.png
Comment (snake14) [inline tests/UI/expected-screenshots/TagManager_manageWebsitesDeleteAction.png:1]: Should this test deleting a site with a container so that we can see the list of links?

**Review by snake14 — CHANGES_REQUESTED** [review]
Looks good overall. I had a couple comments.
When I go to delete a site, it shows the enter password string twice:
image


__Edit:__
It took rebuilding the Vue files for SitesManager, but I got the view to load as expected. So, functional testing passed :+1:

File: tests/UI/expected-screenshots/TagManager_manageWebsitesDeleteAction.png
Comment (adamreng) [reply tests/UI/expected-screenshots/TagManager_manageWebsitesDeleteAction.png:1]: This is what I was going to ask about too. Having a test with a list of containers.

**Review by adamreng — APPROVED** [review]
Functionality works as expected. It even gracefully handles edge case of 100 containers in the "will be deleted" list. Overall implementation looks good 👍

File: tests/UI/expected-screenshots/TagManager_manageWebsitesDeleteAction.png
Comment (AltamashShaikh) [reply tests/UI/expected-screenshots/TagManager_manageWebsitesDeleteAction.png:1]: @snake14 @araichyk  This test will fail once we merge the related core code

Comment (AltamashShaikh) [reply tests/UI/expected-screenshots/TagManager_manageWebsitesDeleteAction.png:1]: It would look like below

image

File: templates/deleteWebsite.twig (hunk 1 above)
Comment (AltamashShaikh) [reply templates/deleteWebsite.twig:6]: @snake14 This was done on purpose to avoid xss being executed
Try creating a new container with name `Test  Imm` and when you try to delete the corresponding site and if we escape only once the xss is executed and with double escaping it doesn't executes at all.

1. With single escape
Screenshot from 2024-08-08 13-51-13

2. With double escape
Screenshot from 2024-08-08 13-48-51

In 1st case the broken image gets displayed which is the probable xss

**Comment by AltamashShaikh** [comment]
@snake14 Did anything changed in SitesManager plugin after vue:build?
> It took rebuilding the Vue files for SitesManager, but I got the view to load as expected. So, functional testing passed

File: templates/deleteWebsite.twig (hunk 1 above)
Comment (snake14) [reply templates/deleteWebsite.twig:6]: Seems like an indication that maybe it should be a Vue template instead of Twig template, but it works :+1:

**Comment by snake14** [comment]
> @snake14 Did anything changed in SitesManager plugin after vue:build?
> 
> > It took rebuilding the Vue files for SitesManager, but I got the view to load as expected. So, functional testing passed

@AltamashShaikh I don't see any change in SitesManager, so it must have just forced the cache even though I already tried clearing the cache via console command :thinking:

**Review by snake14 — APPROVED** [review]
Looks good and functional testing worked as expected :+1:

**Review by snake14 — APPROVED** [review]
Looks good. Confirmed that the message displays as expected while deleting a site from my local instance :+1:

File: lang/en.json [hunk 2]
```diff
@@ -1071,6 +1071,10 @@
         "ClickHtmlAttributeAttributeDescription": "The name of the attribute, such as title, value, alt, or any other HTML attribute.",
         "ClickHtmlAttributeAttributePlaceholder": "e.g. value",
         "PausedTag": "Fantastic! The tag has been paused successfully.",
-        "ResumedTag": "Fantastic! The tag has been resumed successfully."
+        "ResumedTag": "Fantastic! The tag has been resumed successfully.",
+        "MatomoConfigurationMatomoTrackBotsDescription": "By default, Matomo doesn't track visits by bots in order to show accurate visitor metrics. Enable this feature to add tracking for bot visits.",
```
Comment (sgiehl) [inline lang/en.json]:
```suggestion
```

Seems that one was duplicated by accident

Comment (AltamashShaikh) [reply lang/en.json]: Thanks, was added by mistakenly, removed :+1:

**Comment by AltamashShaikh** [comment]
@snake14 Core PR is merged, can you review the PR once again ?

**Review by snake14 — APPROVED** [review]
Looks good. I updated to the latest version of Core and this branch. Everything displayed as expected :+1:
